frontend/src/pages/admin/Settings/General.vue

```diff
@@ -147,20 +171,47 @@ export default {
                     options[s] = this.input[s]
                 }
             })
-
+            // don't save the T&C's date
+            delete options['user:tcs-date']
             // don't save the T&C's URL if no requirement for T&Cs
-            options['user:tcs-url'] = this.input['user:tcs-required'] ? options['user:tcs-url'] : null
+            options['user:tcs-url'] = this.input['user:tcs-required'] ? (options['user:tcs-url'] || '').trim() : null
 
             settingsApi.updateSettings(options)
                 .then(() => {
                     this.$store.dispatch('account/refreshSettings')
+                    this.input['user:tcs-date'] = this.settings['user:tcs-date']
                 })
                 .catch((err) => {
                     console.warn(err)
                 })
                 .finally(() => {
                     this.loading = false
                 })
+        },
+        async updateTermsAndConditions () {
+            // don't save the T&C's if not required
+            if (!this.input['user:tcs-required']) {
+                return
+            }
+            Dialog.show({
+                header: 'Update Terms and Conditions',
+                kind: 'danger',
+                html: '<p>This action will update the timestamp associated with the Terms and Conditions effectively causing existing users to be prompted to re-review and reaccept the updated Terms and Conditions.</p><p>Are you sure?</p>',
```

Comment:
```suggestion
                html: '<p>This action will require all existing users to reaccept the Terms and Conditions the next time they access the platform.</p><p>Are you sure?</p>',
```

I think this message can be simplified. The fact it's a timestamp is an internal implementation detail.

Reply:
much better